### DISCUSSION TITLE
Loading Codable Representation of Navigation Stack with TCA

### DISCUSSION THREAD

**Discussion by giorgioETC** [pointfreeco/swift-composable-architecture#3312] [Q&A]
I have a need to save and load a cached navigation stack during my application lifecycle. I'm new to TCA and am wondering if there is a way to achieve this? I've gone through the docs and past discussions and haven't seen anything regarding this. I had my screens implemented in an MVVM style using the codable property on NavigationPath and would load the cached stack using onAppear, but we want to rewrite our screens using TCA for testability and to add clarity to the complex state we are dealing with.

**Comment by mbrandonw** [comment]
Hi @giorgioETC, I personally would not recommend encoding/decoding the entire navigation `StackState`. The state stored in features typically has a lot of ephemeral things that should not be persisted/restored, such as the loading state of an effect, etc.

Typically you would have a simpler data type that represents the bare essentials of data that can be persisted so that when restored you can construct a reason version of the `StackState`. I understand that you get the codability behavior with `NavigationPath` "for free", but really the cost is that there is not integration of the features and it's not possible to test. And so the cost of recovering tight integration and testing is that you have to do a bit more work for codability.
